SwaMe/RTGrouper.cs

```diff
@@ -47,7 +47,8 @@ public RTMetrics DivideByRT(MzmlParser.Run run, int division, double RTDuration)
 
             for (int i = 0; i < division; i++)
             {
-                RTsegs[i] = run.BasePeaks.First().BpkRTs[0] + RTsegment * i;
+
+                RTsegs[i] = run.Ms2Scans.OrderBy(x=>x.ScanStartTime).First().ScanStartTime + RTsegment * i;

```

Comment:
Sorting is a computationally intensive task and this is being run in a loop - can you move the orderby out of the loop please